src/sync/impl/sync_metadata.cpp

```diff
@@ -121,6 +123,27 @@ SyncMetadataManager::SyncMetadataManager(std::string path,
                 table->set_string(idx_url, entry.get_index(), url.is_null() ? "" : url);
             }
         }
+        if (old_realm->schema_version() < 3) {
+            TableRef table = ObjectStore::table_for_object_type(realm->read_group(), c_sync_userMetadata);
+            size_t idx_identity = table->get_column_index(c_sync_identity);
+            size_t idx_local_uuid = table->get_column_index(c_sync_local_uuid);
+            size_t idx_device_unique_uuid = table->get_column_index(c_sync_device_unique_uuid);
+
+            // Get all the SyncUserMetadata objects.
+            Results results(realm, *table);
+            for (size_t i = 0; i < results.size(); i++) {
+                // Set the device local UUID, but only if the local UUID and the identity are identical.
```

Comment:
My brain's feeling a little mushy right now, but this doesn't seem to describe what the code below is actually doing?

Reply:
Yes, you're right. That comment does not describe the implementation I actually went with; I'll fix it :/.